src/quick-lint-js/parse.h

```diff
@@ -3392,6 +3409,26 @@ class parser {
     bool old_value_;
   };
 
+  class depth_guard {
+   public:
+    explicit depth_guard(parser *p, int old_depth_value) noexcept
+        : parser_(p), old_depth_value_(old_depth_value) {
+      p->depth_++;
+      if (p->depth_ > p->stack_limit) {
+        p->crash_on_unimplemented_token(__FILE__, __LINE__, __func__);
```

Comment:
Almost-blocking: `crash_on_unimplemented_token` seems like the wrong function to call here. I think the error message will be confusing to users. Did you mean to call `crash_on_depth_limit_exceeded` here?

Almost-blocking: Passing this `__FILE__`, `__line__`, and `__func__` is kinda pointless IMO. Did you mean for the caller to give us this information? If we make a separate function, I'm not sure it's worth keeping line information.